src/PushCommand.php

```diff
@@ -400,6 +412,56 @@ private function getPackageName(InputInterface $input = null)
         }
     }
 
+    /**
+     * @param InputInterface $input
+     */
+    private function parseNexusExtra(InputInterface $input){
```

Comment:
To ensure idempotence and to be able to test this function easily in the future, it should return the value of the configuration instead of filling a member variable of the class.
If you want to avoid having to parse those values multiple time, use the member value as cache, and do something like:
```php
if (empty($this->extras)) {
  $this->extras = ...;
}
return $this->extras;
```
This way the code is encapsulated.

It will also allow the possibility to push to multiple repositories at once in the future.

Reply:
If you cache $this - > extras, each caller needs to determine whether the current configuration is the previous unique repository configuration or multi repository configuration. If multi repository configuration is used , caller need to compare “name“ to determine which repository configuration to use. I don't think it's a good idea。

Reply:
The `$extras` was just an example. The current issue is that we have to call `$this->checkNexusPushValid($input);` before being able to call `$this->getNexusExtra()`, which is confusing.
By calling `$this->getNexusExtra()` you expect to receive the value from the configuration directly.

Precisely, the confusing part start here:
https://github.com/Elendev/nexus-composer-push/blob/30af146a8a7303434c8c61aa429891dba7e033f8/src/PushCommand.php#L113
This method is called without returning anything and doesn't seem to do anything, but in reality it does have side-effect on every other calls.

Calling the `sendFile` methods or `generateUrl` from somewhere else in the code withouth calling first `parseNexusExtra` might not return the expected result => there is no idempotence.

In this case, I recommend to get the `nexusExtra` configuration from the first call to `parseNexusExtra` (which can be renamed to `getRepositoryConfiguration($name = null)` and you get the `name` directly from the `execute` command when you call the method).
You can then merge this configuration with whatever is given in parameter through the command line, and then use the final configuration to call every other methods with the correct parameters, avoiding calls to `getNexusExtra` from other methods.

But it's not an easy refactoring, so if you do not have the time to do it, add a comment on the `getNexusExtra` method to specify that the method `parseNexusExtra` has to be called before being able to call it.

Reply:
I see what you mean. I've added some tips. When I'm free, I'll try to refactor the code and discuss it with you then.